### ISSUE TITLE
Build Python ABI3 wheels

### ISSUE THREAD

**Issue by jetuk** [pywr/pywr-next#96]
The updates to the Python extension in #73 are not ABI3 compatible. Building ABI3 wheels would reduce the CI to one build per platform rather requiring individual Python version wheels. 

Currently this is blocked by the extension module using `PyDate` and `PyDateAccess` which are not part of the ABI. Removing these objects from the extension can be done by either:

1. Using `chrono` and the support in `pyo3` for converting from Python datetime objects to `chrono` types, or
2. Replicating the above support but for the `time` crate that is currently used here. One would have to follow the same idea as for the `chrono` conversions in `pyo3` and use the Python API directly in the extension.

**Comment by jetuk** [comment]
PyO3 documentation here: https://pyo3.rs/v0.20.2/building_and_distribution.html#py_limited_apiabi3

**Comment by jetuk** [comment]
Once #115 is merged (1) above can be done, and we should be able to produce ABI3 wheels.

**Comment by jetuk** [comment]
Turns out, it can't be done just yet because the features required in PyO3 haven't been released yet.
